src/common/src/util/value_encoding/row_encoding.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Value encoding is an encoding format which converts row into a binary form that remains
```

Comment:
I guess we should emphasize the "column-aware" in the docs and the namings. "value encoding" should be used to refer to the plain format where each datum is encoded sequentially.

Same for the name of this file. 🤔

```suggestion
//! Column-aware row encoding is an encoding format which converts row into a binary form that remains
```